guiEditor/src/diagram/workbench.tsx

```diff
@@ -65,44 +63,49 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
 
     constructor(props: IWorkbenchComponentProps) {
         super(props);
-        props.globalState.onSelectionChangedObservable.add(selection => {  
-            if(!this._ctrlKeyIsPressed && selection != null)
-            {
-                this.selectedGuiNodes.forEach(element => {
-                element.isSelected = false;
+        props.globalState.onSelectionChangedObservable.add((selection) => {
+            if (!this._ctrlKeyIsPressed && selection != null) {
+                this.selectedGuiNodes.forEach((element) => {
+                    element.isSelected = false;
                 });
-            } 
+            }
             if (!selection) {
                 this._selectedGuiNodes = [];
-            } 
-            else {
-                if (selection instanceof GUINode ) {
+            } else {
+                if (selection instanceof GUINode) {
                     if (this._ctrlKeyIsPressed) {
                         if (this._selectedGuiNodes.indexOf(selection) === -1) {
                             this._selectedGuiNodes.push(selection);
                         }
-                    } 
-                    else {              
+                    } else {
                         this._selectedGuiNodes = [selection];
-                    }              
-                } 
+                    }
+                }
             }
         });
 
         this.props.globalState.hostDocument!.addEventListener("keyup", () => this.onKeyUp(), false);
-        this.props.globalState.hostDocument!.addEventListener("keydown", evt => {         
-            this._ctrlKeyIsPressed = evt.ctrlKey;
-        }, false);
-        this.props.globalState.hostDocument!.defaultView!.addEventListener("blur", () => {
-            this._ctrlKeyIsPressed = false;
-        }, false);     
+        this.props.globalState.hostDocument!.addEventListener(

```

Comment:
seems like this is mainly formatting and not a code change, but - are those events being released at a later time?

Reply:
Like this? Seems like a lot of code. Is there a better way to do this?

Reply:
That won't work, as the function you are passing is not the same object as the one passed to the addEventListener. To get that to work you will need to define local functions that can be passed as a variable to both add and removeEventListener